bigbio/dataloader.py

```diff
@@ -0,0 +1,323 @@
+##!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Meta dataloader to load "load configurations" for all available datasets according to `_SUPPORTED_TASKS`
+"""
+
+import importlib
+import os
+from collections import defaultdict
+from dataclasses import dataclass
+from glob import glob
+from types import ModuleType
+from typing import Optional, Union
+
+import datasets
+from loguru import logger
+
+from bigbio.utils.constants import Tasks
+
+_SCHEMA_TO_TASKS: dict = {
+    "bigbio_kb": [
+        Tasks.NAMED_ENTITY_RECOGNITION,
+        Tasks.NAMED_ENTITY_DISAMBIGUATION,
+        Tasks.EVENT_EXTRACTION,
+        Tasks.RELATION_EXTRACTION,
+        Tasks.COREFERENCE_RESOLUTION,
+    ],
+    "bigbio_qa": [Tasks.QUESTION_ANSWERING],
+    "bigbio_te": [Tasks.TEXTUAL_ENTAILMENT],
+    "bigbio_pairs": [Tasks.SEMANTIC_SIMILARITY],
+    "bigbio_t2t": [Tasks.PARAPHRASING, Tasks.TRANSLATION, Tasks.SUMMARIZATION],
+    "bigbio_text": [Tasks.TEXT_CLASSIFICATION],
+}
+
+
+# NOTE:
+# huge datasets are those for which loading takes >5 minutes
+_HUGE: dict = {
+    "biomrc": [
+        "biomrc_large_B_bigbio_qa",
+        "biomrc_large_A_bigbio_qa",
+    ],
+    "medal": ["medal_bigbio_kb"],
+    "meddialog": ["meddialog_zh_bigbio_text"],
+    "pubtator_central": [
+        "pubtator_central_bigbio_kb",
+    ],
+}
+
+
+# NOTE:
+# weak datasets are the unstructured ones,
+# i.e. not providing a clear task but being more of a resource,
+# e.g. PubTator and MIMICIII
+_WEAK: dict = {
+    "pubtator_central": [
+        "pubtator_central_sample_bigbio_kb",
+        "pubtator_central_bigbio_kb",
+    ]
+}
+
+
+class BigBioDataloader:
+    """
+    Meta dataloader biodatasets
+    """
+
+    def __init__(
+        self,
+        tasks: Optional[list[Tasks]] = None,
+        data_dir: Optional[str] = None,
+        skip_huge: bool = True,
+        skip_weak: bool = True,
+        skip_local: bool = True,
+        exclude: Optional[list] = None,
+    ):
+        self.scripts = sorted(glob(os.path.join("biodatasets", "*", "*.py")))
```

Comment:
this will only work from within the package but we'd like it to work anywhere someone is using the package. maybe try something like, 
```
path_to_here = pathlib.Path(__file__).parent.absolute()                                                                                                                                                                           
self.path_to_biodatasets = path_to_here / "../" / "biodatasets"                                                                        
self.scripts = sorted(glob(os.path.join(self.path_to_biodatasets, "*", "*.py"))) 
```